docs/source/developer/xtensor_internals.rst

```diff
@@ -0,0 +1,21 @@
+.. Copyright (c) 2016, Johan Mabille, Sylvain Corlay and Wolf Vollprecht
+
+   Distributed under the terms of the BSD 3-Clause License.
+
+   The full license is in the file LICENSE, distributed with this software.
+
+xtensor internals
+=================
+
+This section provides information about `xtensor`'s internals and its architecture. It is intended for developers
+who want to contribute to `xtensor` or simply understand how it works under the hood. `xtensor` makes heavy use
+of the CRTP pattern, template programming, universal references and perfect forwarding. One should be familiar with
```

Comment:
template meta-programming.

perfect forwarding -> move semantics?

Reply:
perfect forwarding includes move semantic, it is the general concept associated with universal references.

Reply:
agreed, I just thought that move semantics was more widely used. This is just me scrolling through the documentation.